### DISCUSSION TITLE
Is it possible for ftINIT() to set the thresholds for gene levels in different range?

### DISCUSSION THREAD

**Discussion by Joyjoyjoyc** [SysBioChalmers/RAVEN#532] [Help me!] [answered]
Hi,

Here is a new question for me. I'd like to know is the function ftINIT() classifying metabolic reactions as ‘on’ or ‘off’ with only setting a threshold in the continuous expression levels of the associated genes? and is it possible to set the thresholds for gene levels in different range? like no expression (TPM<a), low expression (a≤TPM<b), medium expression (a≤TPM<c), and high (TPM≥c)(a,b,c are all numbers). cause as I set a vector for transcrData.threshold, there is something wrong. 

really appreciated for your reply!

**Comment by JonathanRob** [comment, marked as answer]
Hi @Joyjoyjoyc,

The `ftINIT` function can handle two different types of input for the `threshold` option:
1) A single value, which applies to all genes
2) A vector of values with the **same size as the number of genes**, allowing different thresholds for each gene

This means that you cannot specify expression ranges (e.g., low, medium, high).

If the expression of a reaction's associated gene is less than the threshold, it will not necessarily turn off the reaction. A weighted score is calculated for each reaction based on the expression of corresponding genes, relative to the threshold. Genes with an expression less than the threshold will have a negative score, and those greater will have a positive score. Moreover, the greater the distance from the threshold, the greater the magnitude of the score (e.g., if the threshold is 1 TPM, a gene expressed at 0.8 TPM will have a less negative score than one at 0.1 TPM). 

The algorithm will then try to maximize the combined score of the reactions kept in the final model. Although this implies that negative-scored reactions are more likely to be removed, it is not always the case. For example, if positive-scored reaction(s) depend on a negative-scored reaction to function (carry flux), it is possible that the reaction will remain in the model even though it has a negative score. Also, if the reaction is necessary to complete a required metabolic task, it will be included in the model regardless of its score.

**Reply by Joyjoyjoyc** [reply]
Really really appreciated for your clear answer!! which helped me a lot in learning GEM. I totally understood now. Thank you so much!!
